pkg/artifacts/helm.go

```diff
@@ -80,38 +80,44 @@ func NewRegistryClientConfig(opts ...RegistryClientOption) *RegistryClientConfig
 }
 
 func getRegistryClientWrap(cfg *RegistryClientConfig) (*registryClientWrap, error) {
-	wrap := &registryClientWrap{}
+	var credentialsFile string
 	opts := []registry.ClientOption{}
+
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: cfg.UseInsecureHTTPS, // #nosec G402
+			},
+		},
+	}
+
 	if cfg.UsePlainHTTP {
 		opts = append(opts, registry.ClientOptPlainHTTP())
 	} else {
-		if cfg.UseInsecureHTTPS { // #nosec G402
```

Comment:
After your change, do we then need this cfg option anymore? If not, then I think it needs to be cleaned up from the codebase.

Reply:
Yes, we need it. It's used here:
```golang
	httpClient := &http.Client{
		Transport: &http.Transport{
			TLSClientConfig: &tls.Config{
				InsecureSkipVerify: cfg.UseInsecureHTTPS, // #nosec G402
			},
		},
	}
````